src/AssistantThreadContextStore.ts

```diff
@@ -0,0 +1,88 @@
+import { MessageMetadataEventPayloadObject } from '@slack/types';
+import { AllAssistantMiddlewareArgs, extractThreadInfo } from './Assistant';
+
+export interface AssistantThreadContextStore {
+  get: GetThreadContextFn;
+  save: SaveThreadContextFn;
+}
+
+export interface GetThreadContextFn {
+  (args: AllAssistantMiddlewareArgs): Promise<AssistantThreadContext>;
+}
+
+export interface SaveThreadContextFn {
+  (args: AllAssistantMiddlewareArgs): Promise<void>;
+}
+
+export interface AssistantThreadContext {
+  channel_id?: string;
+  team_id?: string;
+  enterprise_id?: string | null;
+}
+
+export class DefaultThreadContextStore implements AssistantThreadContextStore {
+  private context: AssistantThreadContext = {};
+
+  public async save(args: AllAssistantMiddlewareArgs): Promise<void> {
+    const { context, client, payload } = args;
+    const { channelId: channel, threadTs: thread_ts, context: threadContext } = extractThreadInfo(payload);
+
+    // Retrieve first several messages from the current Assistant thread
+    const thread = await client.conversations.replies({
+      channel,
+      ts: thread_ts,
+      oldest: thread_ts,
+      include_all_metadata: true,
+      limit: 4,
+    });
+
+    if (!thread.messages) return;
+
+    // Find and update the initial Assistant message with the new context to ensure the
+    // thread always contains the most recent context that user is sending messages from.
+    const initialMsg = thread.messages.find((m) => !m.subtype && m.user === context.botUserId);
+    if (initialMsg) {
+      const { ts, text, blocks } = initialMsg as any; // TODO : TS
```

Comment:
@filmaj If you have any guidance on how to navigate this particular `MessageElement` type issue, would love to hear it. Else I'm likely to just be extra with additional checks.

Reply:
One solution would be something like this. Having type casting for the "blocks" is inevitable: https://github.com/seratch/bolt-js/commit/0d4b52be2fdae38a2be47d09ba8646d6b6d76d35

Reply:
Will give that a try. The Message situation has been a headache from start to finish on this feature.